### PR TITLE
REF: Make WeekOfMonthMixin a cdef class

### PR DESCRIPTION
**Pull request by jbrockmendel**

### CODE REVIEW

**Comment by jreback** [comment]
can you rebase

**Comment by jbrockmendel** [comment]
Rebased +green

**Comment by jbrockmendel** [comment]
rebased+pseudo-green

File: pandas/tseries/offsets.py [hunk 1]
```diff
@@ -299,8 +299,53 @@ def is_on_offset(self, dt):
         # TODO, see #1395
         return True
 
+    def _repr_attrs(self) -> str:
+        # The DateOffset class differs from other classes in that members
```
Comment (jreback) [inline pandas/tseries/offsets.py:303]: this seems like it could be integrated to the other classes no? surely a follown

**Review by jreback — APPROVED** [review]

File: pandas/tseries/offsets.py (hunk 1 above)
Comment (jbrockmendel) [reply pandas/tseries/offsets.py:303]: yah, there are a lot of `__dict__`-related workarounds that can be cleaned up once _everything_ is a cdef class

**Pull request merged**